source/dev_guide/project-standards.rst

```diff
@@ -0,0 +1,154 @@
+
+Run individual hook
+   ::
+
+        # Available hook ids: trailing-whitespace, end-of-file-fixer, check-yaml, black, isort, flake8, mypy
+        pre-commit run <hook_id>.
+
+Squash and Rebase Commits
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before you merge a support branch back into ``master``, the branch is typically
+squashed down to a single buildable commit, and then rebased on top of the upstream ``master`` branch.
+
+-  Ensures build passes from the commit
+-  Cleans up Git history for easy navigation
+-  Makes collaboration and review process more efficient
+-  Makes handling conflicts from rebasing simple since you only have to
+   deal with conflicted commits
+
+Source:
+https://blog.carbonfive.com/always-squash-and-rebase-your-git-commits/
+
+Code Quality Assurance
+----------------------
+
+This project uses several tools for code formatting, linting, and type checking listed below.
+
+- Code Formatting: `black <https://black.readthedocs.io/en/stable/>`__
+- Linting: `flake8 <https://github.com/PyCQA/flake8#flake8>`__, `isort <https://pycqa.github.io/isort/>`__
+- Optional Type Checking: `mypy <http://mypy-lang.org/>`__
+
+You can run them as hooks manually/automatically when committing using ``pre-commit``, or manually through the terminal or IDE/text editor.
+
+Helpful Commands
+~~~~~~~~~~~~~~~~
+
+Run a tool
+    ::
+
+       # Available tool names: black, flake8, isort, mypy
+       <tool_name> .
+
+.. _ci-cd:
+
+Continuous Integration / Continuous Delivery (CI/CD)
+----------------------------------------------------
+
+This project uses `GitHub Actions <https://github.com/E3SM-Project/e3sm_diags/actions>`_ to run a CI/CD workflow on ``push`` and ``pull-request`` to ``master``.
+
+This workflow has three jobs:
+
+1. Run ``pre-commit`` for formatting, linting, and type checking
+2. Run test suite in a conda environment
+3. Publish to Anaconda using a conda environment (Only if #2 succeeds and tagging a release)
+
+Style Guide
+-----------
+
+This project follows the Black code style. Please read about it more `here <https://black.readthedocs.io/en/stable/the_black_code_style.html>`__.
+
+API Documentation
+-----------------
+
+In most cases, code should be self-documenting.
+
+If necessary, documentation should explain **why** something is done, its purpose, and its goal.
+The code shows **how** it is done, so commenting on this can be redundant.
+
+Guidelines
+~~~~~~~~~~
+
+-  Embrace documentation as an integral part of the overall
+   development process
+-  Treat documenting as code and follow principles such as *Don't
+   Repeat Yourself* and *Easier to Change*
+-  Use comments and docstrings to explain ambigiuity, complexity,
+   or to avoid confusion
+-  Co-locate API documentation with related code
+-  Use Python type annotations and type comments where helpful
+
+Things to Avoid
+~~~~~~~~~~~~~~~
+
+-  Write comments as a crutch for poor code
+-  Comment *every* function, data structure, type declaration
```

Comment:
I think this might be better with "Writing..." and "Commenting..."

It's relatively easy to miss that this is a "AVOID" list, and the "Write/Comment" form makes it sound like a to-do list. (At least, that's how it reads to me)

Reply:
I can see how that might be confusing. I had **Don't** to prefix the sentence but removed it. I'll add it back in.